From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 5 Mar 2016 13:26:30 +0100
Subject: [PATCH] {BUG} drm: amdgpu: fix ptr-to-int cast
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc output:

  CC [M]  drivers/gpu/drm/amd/amdgpu/gfx_v7_0.o
drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c: In function ‘gfx_v7_0_ring_emit_vm_flush’:
drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c:3631:17: error: initialization makes integer from pointer without a cast [-Werror=int-conversion]
  uint32_t seq = ring->fence_drv.sync_seq;
                 ^
cc1: all warnings being treated as errors

... sync_seq is an array of integers, not an integer.

Nevertheless I am NOT sure of the fix, taken from gfx_v8_0.c, as the v7
does not seem to use ring->idx.
---
 drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c b/drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c
index 06602df707f8..bf7952f828ef 100644
--- a/drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c
@@ -3628,7 +3628,7 @@ static void gfx_v7_0_ring_emit_vm_flush(struct amdgpu_ring *ring,
 					unsigned vm_id, uint64_t pd_addr)
 {
 	int usepfp = (ring->type == AMDGPU_RING_TYPE_GFX);
-	uint32_t seq = ring->fence_drv.sync_seq;
+	uint32_t seq = ring->fence_drv.sync_seq[ring->idx] /* BUG, not sure of the fix */;
 	uint64_t addr = ring->fence_drv.gpu_addr;
 
 	amdgpu_ring_write(ring, PACKET3(PACKET3_WAIT_REG_MEM, 5));
-- 
